src/NServiceBus.Metrics.PerformanceCounters/PerformanceCounterUpdater.cs

```diff
@@ -1,38 +1,37 @@
 ﻿using System.Collections.Generic;
-using Newtonsoft.Json.Linq;
+using System.Linq;
+using NServiceBus;
 
 class PerformanceCounterUpdater
 {
-    public PerformanceCounterUpdater(PerformanceCountersCache cache, Dictionary<string, CounterInstanceName?> legacyInstanceNameMap)
+    public PerformanceCounterUpdater(PerformanceCountersCache cache, Dictionary<string, CounterInstanceName?> legacyInstanceNameMap, string endpointName)
     {
         this.legacyInstanceNameMap = legacyInstanceNameMap;
+        this.endpointName = endpointName;
         this.cache = cache;
     }
 
-    public void Update(string payload)
+    public void Observe(ProbeContext context)
     {
-        var rootObject = JObject.Parse(payload);
-
-        var context = rootObject.Value<string>("Context");
-        var meters = rootObject["Meters"]?.ToObject<List<Meter>>() ?? new List<Meter>();
-        foreach (var meter in meters)
+        context.Signals.ToList().ForEach(sp =>
         {
             CounterInstanceName? instanceName;
-            legacyInstanceNameMap.TryGetValue(meter.Name, out instanceName);
+            legacyInstanceNameMap.TryGetValue(sp.Name, out instanceName);
+
+            var performanceCounterInstance = cache.Get(instanceName ?? new CounterInstanceName(sp.Name, endpointName));
 
-            var performanceCounterInstance = cache.Get(instanceName ?? new CounterInstanceName(meter.Name, context));
-            performanceCounterInstance.RawValue = meter.Count;
-        }
+            sp.Register(() => performanceCounterInstance.Increment());
+        });
 
-        var timers = rootObject["Timers"]?.ToObject<List<Timer>>() ?? new List<Timer>();
-        foreach (var timer in timers)
+        context.Durations.ToList().ForEach(dp =>
         {
-            var performanceCounterInstance = cache.Get(new CounterInstanceName(timer.Name, context));
-            
-            performanceCounterInstance.RawValue = timer.Histogram.LastValue / 1000;
-        }
+            var performanceCounterInstance = cache.Get(new CounterInstanceName(dp.Name, endpointName));
+
+            dp.Register(d => performanceCounterInstance.RawValue = (long)d.TotalSeconds);
```

Comment:
The base must is incremented, but the processed time should be added. I don't think the raw value must be set here.

https://manski.net/2013/11/windows-performance-counter-types/#average-counters

> Average time each database transaction takes; for each transaction, you’d measure the execution time of the transaction with Stopwatch and increment AverageTimer32 by Stopwatch.ElapsedTicks; and for each completed transaction you’d increment AverageBase by one